corehq/apps/commtrack/sms.py

```diff
@@ -90,9 +91,9 @@ def __init__(self, domain, v):
                 raise NotAUserClassError
 
             # currently only support one location on the UI
-            linked_loc = u.location
-            if linked_loc:
-                self.location = get_supply_point(self.domain.name, loc=linked_loc)
+            self.location = u.location
```

Comment:
Originally throughout this code `self.location` was a dict `{'case': supply_point, 'location': location}`. This PR basically separates that out into `self.location` and `self.case` and then tidies up the functions that fetch the location and the case.